topk-js/src/data/value.rs

```diff
@@ -0,0 +1,244 @@
+use std::ptr;
+
+use napi::{
+    bindgen_prelude::*,
+    sys::{napi_is_array, napi_is_buffer},
+};
+
+use napi_derive::napi;
+
+use super::{
+    utils::{get_napi_value_type, is_napi_integer},
+    vector::{Vector, VectorUnion},
+};
+
+#[derive(Debug, Clone)]
+pub enum Value {
+    String(String),
+    Bool(bool),
+    F64(f64),
+    U32(u32),
+    U64(u64),
+    I32(i32),
+    I64(i64),
+    F32(f32),
+    Binary(Vec<u8>),
+    Vector(Vector),
+    Null,
+}
+
+#[napi(namespace = "data")]
+pub fn binary(values: Vec<u8>) -> Value {
+    Value::Binary(values)
+}
+
+impl From<Value> for topk_protos::v1::data::Value {
+    fn from(value: Value) -> Self {
+        match value {
+            Value::String(s) => topk_protos::v1::data::Value::string(s),
+            Value::F64(n) => topk_protos::v1::data::Value::f64(n),
+            Value::Bool(b) => topk_protos::v1::data::Value::bool(b),
+            Value::U32(n) => topk_protos::v1::data::Value::u32(n),
+            Value::U64(n) => topk_protos::v1::data::Value::u64(n),
+            Value::I32(n) => topk_protos::v1::data::Value::i32(n),
+            Value::I64(n) => topk_protos::v1::data::Value::i64(n),
+            Value::F32(n) => topk_protos::v1::data::Value::f32(n),
+            Value::Binary(b) => topk_protos::v1::data::Value::binary(b),
+            Value::Vector(v) => v.into(),
+            Value::Null => topk_protos::v1::data::Value::null(),
+        }
+    }
+}
+
+impl From<topk_protos::v1::data::Value> for Value {
+    fn from(value: topk_protos::v1::data::Value) -> Self {
+        match value.value {
+            Some(topk_protos::v1::data::value::Value::String(s)) => Value::String(s),
+            Some(topk_protos::v1::data::value::Value::F64(n)) => Value::F64(n),
+            Some(topk_protos::v1::data::value::Value::Bool(b)) => Value::Bool(b),
+            Some(topk_protos::v1::data::value::Value::U32(n)) => Value::I32(n.try_into().unwrap()),
```

Comment:
this will fail if number can't be converted. let's replace `.unwrap()` with `.expect()` so at least there's a good error message